File: native/src/knapsack.cpp
#include "knapsack.hpp"

// add an item to the pool of items to be considerated for knapsack problem
void Knapsack::addItem(int weight, int value) 
{
	Item nItem;
	nItem.weight = weight;
	nItem.value = value;
	items.push_back(nItem);
}

// set capacity of knapsack
void Knapsack::setCapacity(int cap)
{
	capacity = cap;
}

// solves the current knapsack problem
ItemList Knapsack::solve() 
{
	// dummy function for testing:
	ItemList mList;
	for (int i = 0; i < items.size(); ++i)
	{
		mList.push_back(i);
	}
	return mList;
}

File: boost/knapsack.py
import libknapsack

def main():

	# setup libknapsack object
    ks = libknapsack.Knapsack()
    ks.setCapacity(5)

    # add items to knapsack
    ks.addItem(0,3,10)
    ks.addItem(1,1,2)
    ks.addItem(2,2,2)
    ks.addItem(3,1,3)

    # print solution
    for item in ks.solve():
        print(item)

if __name__ == "__main__":
    main()